[Jazhten Low] iP

jazhten opened the pull request.

jazhten committed:
Add basic Duke echo reply functionality

jazhten committed:
Add the ability to store text entered by user and display back when requested

jazhten committed:
Add the ability to mark tasks as done.

jazhten committed:
Add Todos
Add Deadlines
Add Events

jazhten committed:
Add Automated Text UI Testing

okkhoy commented:
If this is your submission for this week's IP task, please edit the title to follow the specified convention: https://nus-cs2113-ay2021s1.github.io/website/schedule/week4/project.html#1-create-a-pr-to-the-upstream-repo-mon-aug-31st-2359

@jazhten for your action, please

jazhten committed:
Improve code quality

weishuangtan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,43 @@
+                    System.out.println("\t"+(i+1) +". "+ storedTasks[i].getTaskInfo());
+                }
+            }
+            else if (input.equals("bye")) {
```
Incorrect bracket layout for if-else statements

weishuangtan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+    public void completeTask(){
+        this.isDone = true;
+    }
+    /*
```
Comments should have a /** opening on a separate line

weishuangtan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,43 @@
+                System.out.println("Bye. May the Force be with you.");
+                break;
+            }
+            else if (splitInput[0].equals("done")){
```
Can consider standardising bracket formats:

eg if (....) { } vs if (...){ }

jazhten replied:
Yes, I hadn't noticed that. Thanks for pointing it out!

weishuangtan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,43 @@
+
 public class Duke {
     public static void main(String[] args) {
         String logo = " ____        _        \n"
```
Can caps LOGO because the string is constant

weishuangtan reviewed (commented):
Overall coding standards are great! Just need to take note of the bracket layouts and follow them.

amanda-chua commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,43 @@
+                int taskNum = Integer.parseInt(splitInput[1])-1;
+                storedTasks[taskNum].completeTask();
+                String task = storedTasks[taskNum].getTaskName();
+                System.out.println("\tNice! I've marked this task as done: ");
```
Can add functions to do printing to make the main shorter.

jialerk replied:
Maybe you can use functions instead of writing everything in the main function :)

amanda-chua reviewed (commented):
LGTM

amanda-chua commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.isDone = true;
+    }
+    /*
+    public String getStatusIcon(){
```
Can remove redundant code here.

amanda-chua commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.isDone = false;
+    }
+
+    public void completeTask(){
```
Maybe can name as markDone instead?

amanda-chua reviewed (commented):
Updated review with more details.

jialerk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,43 @@
+            String[] splitInput = input.split(" ");
+            if(input.equals("list")){
+                System.out.println("\tHere are the tasks in your list:");
+                for(int i=0;i<index;i++){
```
might be an arrowhead code, maybe you can make the printing of tasks into a function instead :)

jazhten replied:
Yes great idea thanks!

jialerk commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,43 @@
+        System.out.println("How may I be at your pleasure today?");
+        Scanner sc = new Scanner(System.in);
+        Task[] storedTasks = new Task[100];
+        int index = 0;
```
Maybe you can make your index naming clearer, like indexOfLastTask

jialerk reviewed (commented):
Good job overall!!

jazhten committed:
Add error handling capabilities

jazhten committed:
Add some spaces for code readability

jazhten committed:
Merge branch 'branch-Level-5'

joelczk commented on src/main/java/InputManager.java:
```diff
@@ -0,0 +1,90 @@
+            this.currentTaskIndex += 1;
+            return;
+
+        case "deadline":
+            String deadlineOperation;
+            try {
+                deadlineOperation = StringOperations.getTimeline(input, cmd);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return;
+            }
+            String deadlineInputDescription = StringOperations.getDescription(input);
+            storedTasks[currentTaskIndex] = new Deadline(deadlineInputDescription, deadlineOperation);
+            StringOperations.printAddResponse(storedTasks[currentTaskIndex], currentTaskIndex + 1);
+            this.currentTaskIndex += 1;
+            return;
+
+        case "event":
+            String eventOperation;
+            try {
+                eventOperation = StringOperations.getTimeline(input, cmd);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return;
+            }
+            String eventInputDescription = StringOperations.getDescription(input);
+            storedTasks[currentTaskIndex] = new Event(eventInputDescription, eventOperation);
+            StringOperations.printAddResponse(storedTasks[currentTaskIndex], currentTaskIndex + 1);
+            this.currentTaskIndex += 1;
+            return;
+            
+        case "":
+            System.out.println(DukeException.ExceptionResponse.EXCEPTION_UNIDENTIFIED_INPUT);
+            return;
+
+        default:
+            System.out.println(DukeException.ExceptionResponse.EXCEPTION_UNIDENTIFIED_INPUT);
+        }
+    }
+}
```
This is too long. Please consider to refactor this.

joelczk commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,25 @@
+public class DukeException extends Exception {
+
+    public enum ExceptionResponse {
+        EXCEPTION_NO_DESCRIPTION("\tMissed the Description you have, young padawan..."),
+        EXCEPTION_NO_TIME_DATA("\tOopsies! Mesa thinksa yousa missed the time"),//jar jar binks btw
+        EXCEPTION_NO_FORMAT_IDENTIFIER("\tThe force requires you provide the correct identifier (/by /at)."),
+        EXCEPTION_COMPLETE_UNDEFINED_TASK("\tToo hasty you are, the task has not been defined yet"),
+        EXCEPTION_UNIDENTIFIED_INPUT("\tThese are not the inputs you are looking for...Please input something");
+
+        private String exception;
+
+        ExceptionResponse(String exceptionMessage) {
+            this.exception = exceptionMessage;
+        }
+
+        @Override
+        public String toString() {
+            return exception;
+        }
+    }
+
+    public DukeException(ExceptionResponse exception) {
+        super(exception.toString());
+    }
+}
```
Good attempt at using Enums!

joelczk commented on src/main/java/StringOperations.java:
```diff
@@ -0,0 +1,93 @@
+    Function to print the response when a task is marked as completed by the user.
+     */
+    public static void completeTask(Task[] tasks, int taskIndex) {
+        System.out.println("\tNice! I've marked this task as done: ");
```
You could consider declaring the constant String as a constant variable in the class instead

joelczk commented on src/main/java/StringOperations.java:
```diff
@@ -0,0 +1,93 @@
+        try {
+            output = input.substring(getFirstSpace(input), startPoint);
+            return output;
+        } catch (Exception e) {
```
you could perhaps give a more meaning variable name instead

joelczk commented on src/main/java/InputManager.java:
```diff
@@ -0,0 +1,90 @@
+            return;
+
+        case "bye":
+            System.out.println("Bye. May the Force be with you.");
```
You could perhaps declare the string as a constant variable for readability.

joelczk reviewed (commented):
Generally well done. Do take note of the comments and make the relevant changes.

jazhten committed:
Add delete function

jazhten committed:
Clean the code

jazhten committed:
Add Save function

jazhten committed:
Merge branch 'branch-Level-6'

jazhten committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/InputManager.java

jazhten committed:
Resolve merge conflicts

jazhten committed:
Minor bug fixes

jazhten committed:
Add Ui
Add Storage
Add Parser
Add TaskList
Fix runtest errors

jazhten committed:
Add find function

jazhten committed:
Add appropriate comments and documentation

jazhten committed:
Merge branch 'branch-A-JavaDoc'

jazhten committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/Ui.java

jazhten committed:
Update readme

jazhten committed:
Set theme jekyll-theme-hacker

jazhten committed:
Attempt to fix readme

jazhten committed:
Merge branch 'master' of https://github.com/jazhten/ip

jazhten committed:
Try to resolve theme branching issue

jazhten committed:
Fix readme formatting

jazhten committed:
Fix formatting of README.md

jazhten committed:
Add DateTime functionalities
Add date function to find all tasks after specified date
Add time function to find all tasks after specified time

jazhten committed:
Add date time search function
Move utility (parser etc) into package

jazhten committed:
Add javaDocs
Modify test files

jazhten committed:
Add help command
